### PR TITLE
[Merged by Bors] - chore(ContinuousMonoidHom): golf

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Algebra/ContinuousMonoidHom.lean [hunk 1]
```diff
@@ -286,47 +286,24 @@ theorem embedding_toContinuousMap :
 #align continuous_monoid_hom.embedding_to_continuous_map ContinuousMonoidHom.embedding_toContinuousMap
 #align continuous_add_monoid_hom.embedding_to_continuous_map ContinuousAddMonoidHom.embedding_toContinuousMap
 
+@[to_additive]
+lemma range_toContinuousMap :
+    Set.range (toContinuousMap : ContinuousMonoidHom A B → C(A, B)) =
+      {f : C(A, B) | f 1 = 1 ∧ ∀ x y, f (x * y) = f x * f y} := by
+  refine Set.Subset.antisymm (Set.range_subset_iff.2 fun f ↦ ⟨map_one f, map_mul f⟩) ?_
+  rintro f ⟨h1, hmul⟩
+  exact ⟨{ f with map_one' := h1, map_mul' := hmul }, rfl⟩
```
Comment (alreadydone) [inline Mathlib/Topology/Algebra/ContinuousMonoidHom.lean:295]:
```suggestion
      {f : C(A, B) | f 1 = 1 ∧ ∀ x y, f (x * y) = f x * f y} :=
  (Set.range_subset_iff.2 fun f ↦ ⟨map_one f, map_mul f⟩).antisymm
    fun f ⟨h1, hmul⟩ ↦ ⟨{ f with map_one' := h1, map_mul' := hmul }, rfl⟩
```
(untested)

File: Mathlib/Topology/Algebra/ContinuousMonoidHom.lean [hunk 2]
```diff
@@ -286,47 +286,24 @@ theorem embedding_toContinuousMap :
...
-        rintro _ ⟨x, y, hx : (x, y).1 ∈ U, hy : (x, y).2 ∈ V, rfl⟩
-        exact h' ⟨hx, hy⟩⟩⟩
+    ClosedEmbedding (toContinuousMap : ContinuousMonoidHom A B → C(A, B)) where
+  toEmbedding := embedding_toContinuousMap A B
+  closed_range := by
+    simp only [range_toContinuousMap, Set.setOf_and, Set.setOf_forall]
+    refine .inter (isClosed_singleton.preimage (ContinuousMap.continuous_eval_const 1)) <|
+      isClosed_iInter fun x ↦ isClosed_iInter fun y ↦ ?_
+    exact isClosed_eq (ContinuousMap.continuous_eval_const (x * y)) <|
+      .mul (ContinuousMap.continuous_eval_const x) (ContinuousMap.continuous_eval_const y)
```
Comment (alreadydone) [inline Mathlib/Topology/Algebra/ContinuousMonoidHom.lean:306]: looks like this can be a single `exact` too.

**Review by alreadydone — APPROVED** [review]

File: Mathlib/Topology/Algebra/ContinuousMonoidHom.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Topology/Algebra/ContinuousMonoidHom.lean:295]: Doesn't work because this way Lean doesn't know the type of `⟨map_one f, map_mul f⟩`.

**Comment by alreadydone** [comment]
I see, thanks!
maintainer merge

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**